tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -1292,6 +1293,36 @@ remove_dead_instrs(void)
   }
 }
 
+/**
+   \brief process debug info of constants with parameter attribute.
+ */
+static void process_params(void) {
+  unsigned smax = stb.stg_avail;
+  for (SPTR sptr = get_symbol_start(); sptr < smax; ++sptr) {
+    if (STYPEG(sptr) == ST_PARAM && should_preserve_param(sptr)) {
+      DTYPE dtype = DTYPEG(sptr);
```

Comment:
can should_preserve_param take dtype as argument? If so you hoist the getting of dtype.

Reply:
Thanks for the comment. It will be incorporated.